feat!: upgrade MUI library to v7 (#439)
eclipse-tractusx/portal-shared-components: docs/admin/migration/migration.md

# Migration guide of shared components

> ## Breaking changes MUI v5 -> v7

Reviewer: provide information about the relevant version of the shared components, for which version upgrade is this information relevant? the currently latest version is 3.7.6, see [releases](https://github.com/eclipse-tractusx/portal-shared-components/releases), so I assume this change should be released as 4.0.0 because you state here that upgrade of the packages is to be considered a breaking change, right?

Author: @evegufy Yes the target version should be 4.0.0, I have updated this in the document. 

This MUI update has been tested with storybook. All breaking changes were fixed and all components are functioning as intended. 

I have also tested against the portal, where there is breaking changes. This is what I am referring to in the migration guide with Typography, PageLoadingTable, Table components. When overriding certain props from these components on portal side, these components break and will need to be updated with the recommendations from the MUI documentation.

Reviewer: doesn't an issue for the integration of the upgrade exist? it should be taken up right after the release of the 4.0.0 package.

Author: @evegufy I don't believe there is currently an issue created for the integration on portal side for 4.0.0. However, I can create the issue and assign myself as the responsible developer. 

Also, thank you for all your migration file improvements, I have committed all your suggestions.

Reviewer: @JordanGerada3 yes, please create the issue in portal-frontend and take it up as soon as the 4.0.0 package has been released.

Author: @evegufy here is the portal-frontend issue for reference: [1565](https://github.com/eclipse-tractusx/portal-frontend/issues/1565)

> # Migration guide of shared components version 4.0.0

Reviewer: ```suggestion
# Migration guide of shared components
```

> ## Breaking changes of upgrade MUI v5 -> v7

Reviewer: ```suggestion
## Breaking changes of upgrade from version 3.0.0 to 4.0.0 (MUI v5 -> v7)
```

> ## Breaking changes of upgrade from version 3.0.0 to 4.0.0 (MUI v5 -> v7)

Reviewer: ```suggestion
## Upgrade from version 3.0.0 to 4.0.0

### Breaking changes MUI v5 -> v7
```

>     When consuming the Shared components from release version 4.0.0, overriding some component props will no longer work as a result of the MUI upgrade.

Reviewer: Not sure why the indents where added, but they cause a copy/paste format in md, which - I assume - isn't intended.

```suggestion
When consuming the Shared components from release version 4.0.0, overriding some component props will no longer work as a result of the MUI upgrade.
```

> ### The following Shared components may be affected by the update:

Reviewer: ```suggestion
#### The following Shared components may be affected by the update:
```

>     Overriding this shared component directly in the target project may cause issues.
>     Please referer to the MUI documentaion https://mui.com/material-ui/migration/upgrade-to-v6 and make the recommended changes.
> 
>     - Typography (Uses MUI Typography)

Reviewer: typos and url linking, besides indentation
```suggestion
Overriding this shared component directly in the target project may cause issues.
Please refer to the [MUI upgrade to v6 documentation](https://mui.com/material-ui/migration/upgrade-to-v6) and make the recommended changes.

  - Typography (Uses MUI Typography)
```

> ## Breaking changes MUI DataGrid v6 -> v7

Reviewer: ```suggestion
### Breaking changes MUI DataGrid v6 -> v7
```

> ### The following Shared components may be affected by the update:

Reviewer: ```suggestion
#### The following Shared components may be affected by the update:
```

>     Overriding these shared component directly in the target project may cause issues.
>     Please referer to the MUI documentaiton: https://mui.com/x/migration/migration-data-grid-v6/ and make the recommended changes.
> 
>     - PageLoadingTable (Uses MUI DataGrid)
>     - Table (Uses MUI DataGrid)

Reviewer: typos, besides indentation
```suggestion
Overriding these shared component directly in the target project may cause issues.
Please refer to the [MUI migration data-grid v6 documentation](https://mui.com/x/migration/migration-data-grid-v6) and make the recommended changes.

- PageLoadingTable (Uses MUI DataGrid)
- Table (Uses MUI DataGrid)
```

> ### MUI migration documention:

Reviewer: ```suggestion
#### MUI migration documentation
```

>     - MUI: https://mui.com/material-ui/migration/upgrade-to-v6
>     - MUI: https://mui.com/material-ui/migration/upgrade-to-v7
>     - MUI DataGrid: https://mui.com/x/migration/migration-data-grid-v6/

Reviewer: ```suggestion
- MUI: <https://mui.com/material-ui/migration/upgrade-to-v6>
- MUI: <https://mui.com/material-ui/migration/upgrade-to-v7>
- MUI DataGrid: <https://mui.com/x/migration/migration-data-grid-v6>
```
